Reflect composition type, cleanup resetting

tomalec opened the pull request:
- Reflect composition kind as an attribute, 
  to improve debugging, app development and composing
- Make resetting cleaner, 
  remove not needed `temporaryComposition` property,
  make `storedLayout` "private": `_storedLayout`

tomalec committed:
Reflect composition kind as an attribute,

to improve debugging, app development and composing

tomalec committed:
Make resetting cleaner,

remove not needed `temporaryComposition` property,
make `storedLayout` "private": `_storedLayout`

tomalec committed:
Remove redundant `});` leftover,

warpech commented on starcounter-include.html:
```diff
@@ -277,82 +277,87 @@
-                                        else {
-                                            return false;
+            // this is a request from composition editor to reset to default composition
+            const forceDefault = (compositionString === ""); // compositionString === null
```
I don't understand the comment: `// compositionString === null`

Do you mean: `// should be changed to compositionString === null` ?

tomalec replied:
> Do you mean: `// should be changed to compositionString === null` ?

Exactly

warpech commented on starcounter-include.html:
```diff
@@ -277,82 +277,87 @@
+            // render temporary composition if given
             if (compositionString) {
                 this.setAttribute('composition', 'temporary');
-                this.temporaryComposition = compositionString;
```
Very clever to remove it. Indeed it was not needed.

warpech commented on starcounter-include.html:
```diff
@@ -411,7 +416,7 @@
                     // and it's different than already stored one
                     subPath === this.compositionProvider + '.Composition' &&
-                    value != this.storedLayout
+                    value != this._storedLayout
```
Maybe you'd like to rename 

- `_storedLayout` -> `_storedComposition`
- `_forceLayoutChange` -> `_forceRenderComposition`

tomalec replied:
I'd be happy to, but I'm not sure   id `_forceLayoutChange` is not used by some other elements.

warpech replied:
It is only used by us https://github.com/search?q=_forceLayoutChange&type=Code. But I agree, let's keep `_forceLayoutChange` for now.

warpech reviewed (changes requested):
Only minor comments

warpech commented:
Note that we must not release this until https://github.com/Starcounter/Blending/pull/500 is merged.

tomalec committed:
Change code comment to be more explanatory

as suggested at https://github.com/Starcounter/starcounter-include/pull/122#discussion_r254884415

tomalec committed:
Rename `_storedLayout` to `_storedComposition`,

as suggested at https://github.com/Starcounter/starcounter-include/pull/122#discussion_r254888904

tomalec committed:
Merge branch 'Blending/issues/469-fix-reset-reflect-composition-type' of github.com:Starcounter/starcounter-include into Blending/issues/469-reflect-composition-type-cleanup-resetting

tomalec commented:
@warpech is it good to go now? https://travis-ci.org/Starcounter/starcounter-include/builds/491577773

warpech commented:
I started a new build, because the last one crashed: https://travis-ci.org/Starcounter/starcounter-include/builds/491577773

I tested the change locally. Looks ok!

tomalec commented:
I'd wait with the release for some thing more significant, unless you think we should push it directly to SCF?
